### PR TITLE
fix: index file judgement bug (#306)

### PR DESCRIPTION
**Pull request by meteorlxy**
#306

### CODE REVIEW

File: lib/prepare.js [hunk 1]
```diff
@@ -236,7 +236,7 @@ async function genComponentRegistrationFile ({ sourceDir }) {
   return `import Vue from 'vue'\n` + components.map(genImport).join('\n')
 }
 
-const indexRE = /\b(index|readme)\.md$/i
+const indexRE = /(?<=(^|\/))(index|readme)\.md$/i
 const extRE = /\.(vue|md)$/
```
Comment (ulivz) [inline lib/prepare.js]: `lookbehind` is good, how about `/(^|\/)(index|readme)\.md$/i`? it's same for this case.

Comment (meteorlxy) [reply lib/prepare.js]: It's not the same. `\b` will not  match the word boundary itself. But `/(^|\/)(index|readme)\.md$/i` will match the begining `/`

Comment (meteorlxy) [reply lib/prepare.js]: I meant  /(^|\/)(index|readme)\.md$/i too... /(^|\/)(index|readme)\.md$/i has different behavior

Comment (meteorlxy) [reply lib/prepare.js]: Well, in other words, `\b` and `(?<=(^|\/))` won't include the begining `/`, but `(^|\/)` will

**Comment by meteorlxy** [comment]
Alternative code:
```js
const indexRE = /(^|\/)(index|readme)\.md$/i

// ...

    // README.md -> /
    // foo/README.md -> /foo/
    const fileReplace = file.replace(indexRE, '')
    return '/' + ( fileReplace === '' ? '' : fileReplace + '/')
```

**Pull request merged**